include/linux/regmap.h

```diff
@@ -499,6 +499,10 @@ struct regmap_range_cfg {
 	unsigned int window_len;
 };
 
+struct regmap_sdw_mbq_cfg {
+	int (*mbq_size)(struct device *dev, unsigned int reg);
+};
```

Comment:
so each codec driver would essentially have a switch table to define the size based on address ranges?

IOW it would be an extension of the writable/readable/volatile handling with one more callback?

Reply:
the entity number can be found based on the address range, it's not complicated with a direct mapping from function, entity, control selector and control number to register address.